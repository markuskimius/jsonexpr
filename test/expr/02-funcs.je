hello = "Hello, world!";

PRINT("== CEIL, FLOOR, ROUND ==");
PRINT(CEIL(3.5));
PRINT(FLOOR(3.5));
PRINT(ROUND(3.5));
PRINT(CEIL(-3.5));
PRINT(FLOOR(-3.5));
PRINT(ROUND(-3.5));
PRINT();

PRINT("== LEN, SQRT ==");
PRINT(LEN(""));
PRINT(LEN([]));
PRINT(LEN({}));
PRINT(LEN("Hello"));
PRINT(LEN([ "Hello", "world!" ]));
PRINT(LEN({ "Hello": "world!" }));
PRINT(SQRT(9));
PRINT(SQRT(3.5));
PRINT(SQRT(-10));
PRINT();

PRINT("== BINARY, HEX ==");
PRINT(BINARY(0xf0f0));
PRINT(HEX(0xf0f0));
PRINT();

PRINT("== EVAL ==");
EVAL("
    PRINT(hello);
");
PRINT();

PRINT("== FUNCTION ==");
FUNCTION("#",
    PRINT(ARGS[0] * ARGS[0])
)(7);
PRINT();

PRINT("== IF ==");
IF(true, PRINT(1));
IF(true, PRINT(1), PRINT(2));
IF(false, PRINT(1), PRINT(2));
IF(false, PRINT(1), true, PRINT(2));
IF(false, PRINT(1), true, PRINT(2), PRINT(3));
IF(false, PRINT(1), false, PRINT(2), PRINT(3));
PRINT();

PRINT("== FOR ==");
FOR(i=0, i<10, i++,
    PRINT(i)
);
PRINT();

PRINT("== FOREACH ==");
FOREACH(i, [ "Hello", "world!" ],
    PRINT(i)
);
PRINT();

FOREACH(i, { "Hello":"world!", "Bye":"cruel world!" },
    PRINT(i[0], i[1])
);
PRINT();

PRINT("== WHILE ==");
i=0; WHILE(i<10,
    PRINT(i);
    i++;
);
PRINT();
