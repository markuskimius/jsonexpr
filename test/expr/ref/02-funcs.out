== CEIL, FLOOR, ROUND ==
4
3
4
-3
-4
-4

== LEN, SQRT ==
0
0
0
5
2
1
3.0
1.8708286933869706
nan

== BINARY, HEX ==
0000000000000000000000000000000000000000000000001111000011110000
000000000000F0F0

== EVAL ==
Hello, world!

== FUNCTION ==
49

== IF ==
1
1
2
2
2
3

== FOR ==
0
1
2
3
4
5
6
7
8
9

== FOREACH ==
Hello
world!

Bye cruel world!
Hello world!

== WHILE ==
0
1
2
3
4
5
6
7
8
9

